autoload/multiple_cursors.vim

```diff
@@ -626,6 +629,40 @@ function! s:CursorManager.restore_user_settings() dict
   call setreg('"', s:paste_buffer_temporary_text, s:paste_buffer_temporary_type)
 endfunction
 
+" Restore the new cursor positions for transition from Normal into Insert mode
+function! s:CursorManager.restore_positions() dict
+  if s:char ==# 'v' && s:to_mode ==# 'n'
+    if s:saved_char ==# 'I'
+      call self.start_loop()
+      for _ in range(self.size())
+        let pos = remove(self.saved_positions, 0)
+        call self.get_current().update_position(pos)
+        call cursor(pos)
+        call self.next()
+      endfor
```

Comment:
Thanks, I like it better! However, I was talking about this very block from `start_loop` to `endfor`. The first `call` and `for` loop contain no dependencies on the logic of the code flow (no conditions on `to_mode` or `char` or `saver_char`) and thus are pretty much general and can be reused anywhere. I like this separation, but I think that loop can be extracted further (as well as similar loop in `save_positions`. Hope I'm not nitpicking too much.